### PR TITLE
Update Spin CLI Reference

### PR DESCRIPTION
**Pull request by tpmccallum**
Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)

Screenshot 2023-03-20 at 2 39 07 pm

- [x] Has run `npm run build-index`
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)

Screenshot 2023-03-20 at 2 37 32 pm

- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors

Screenshot 2023-03-20 at 2 50 12 pm
Screenshot 2023-03-20 at 2 45 22 pm

### CODE REVIEW

**Review by michelleN — APPROVED** [review]

File: content/common/cli-reference.md
Comment (vdice) [inline content/common/cli-reference.md:1]: Hi @tpmccallum, as I previewed this page, I noticed a few items that we may want to address.  Any/all that you agree are worth visiting can be follow-ups.  (Some aren't directly related to these changes, so definitely feel free to prioritze for a later date.)

1. Do we want to update this note to "In the Spin v1.0.0 release..."?
developer-488-1

---
2. Should this entry be removed now that the command has been fully removed in v1.0.0? 
developer-488-2

---

3. I believe we are considering OCI Registry support no longer experimental -- also the command output from `spin registry --help` looks to need updating (in v1.0.0, the command text also dropped the experimental mentions).
developer-488-4

(A reminder to update the ToC link accordingly:)
developer-488-3

---
4. It looks like extra `\`'s are making there way to the rendered entries here.  For example, I'm assuming `spin plugins <install\|list\|uninstall\|update\|upgrade>` is actually intended to show up as `spin plugins <install|list|uninstall|update|upgrade>`?
developer-488-5

**Review by vdice — COMMENTED** [review]
Looking great!

File: content/common/cli-reference.md
Comment (michelleN) [reply content/common/cli-reference.md:1]: good catches!

Comment (tpmccallum) [reply content/common/cli-reference.md:1]: +1 
Good catches! @vdice 
All updated in new push to this branch.
Thanks @michelleN and @vdice

**Pull request merged**